static/js/show-task.js

```diff
@@ -39,20 +41,56 @@ function saveTask() {
     }});
 }
 
+function rememberState() {
+	// Finding how many number of rows are being displayed so that we can make it same after refresh
+        var dropdownElements = $('.page-list .dropdown-menu .dropdown-item');
+        var activeDropdownElement = $('.page-list .dropdown-menu .dropdown-item.active')[0];
+        for(var i=0; i < dropdownElements.length; i++)
+        {
+            if (dropdownElements[i] === activeDropdownElement) {
+                activeDropdownElementIndex = i;
+            }
+        }
+
+        // Finding which page is active so that we can make it active again after refresh
+        var paginationElements = $('.pagination .page-item .page-link');
+        var activeElement = $('.pagination .page-item.active .page-link')[0];
+        for(var i=0; i < paginationElements.length; i++)
+        {
+            if (paginationElements[i] === activeElement) {
+                activeElementIndex = i;
+            }
+        }
+}
+
+function restoreState() {
+      // making the number of rows to be same
+      var dropdownElements = $('.page-list .dropdown-menu .dropdown-item');
+      dropdownElements[activeDropdownElementIndex].click();
+
+      // making the same page active as was before
+      var paginationElements = $('.pagination .page-item .page-link');
+      paginationElements[activeElementIndex].click();
+}
+
+
 function refresh() {
     search_term = $("#search_by_task_name_form input[name=search_term]").val();
     worklist_name = $("#search_by_task_name_form input[name=worklist_name]").val();
     worklist_created_by = $("#search_by_task_name_form input[name=worklist_created_by]").val();
 
+    rememberState();
+
     $('#task_table').html('');
     $('#task_table').addClass('loader');
     $('#task_table').html('').load(
-    "/worklist-tool/update-task-table?search_term=" + encodeURIComponent(search_term) + "&worklist_name=" +
-        encodeURIComponent(worklist_name) + "&worklist_created_by=" + encodeURIComponent(worklist_created_by),
-    function() {
-      $('#task_table').removeClass('loader');
-    }
+    "/worklist-tool/update-task-table/" + encodeURIComponent(worklist_created_by) + "/" + encodeURIComponent(worklist_name) + "?search_term=" + encodeURIComponent(search_term),
+	    function() {
+	      $('#task_table').removeClass('loader');
+	    }
     );
+
+    setTimeout(restoreState, 2000);
```

Comment:
Why do this on a timeout rather than when the request completes?

Reply:
I tried that only but the problem was that dropdownElements and paginationElements weren't defined by then. This is because bootstrap-table elements are reloaded only after the load function is fully executed. Hence, I had to approximate the time the function might take to execute (for now 2 seconds) and then set a timeout accordingly.

Reply:
Yeah, the fixed timeout is pretty brittle, let's try to do better. How do the bootstrap-table elements get reloaded? That is, we load the new table HTML from the server, overwrite the contents of `#tasks_table`, but never call any bootstrap-table methods on it, so presumably bootstrap-table has some event listener set up to reload itself?

If so, wonder if we can rely on some of bootstrap-table's [events](http://bootstrap-table.wenzhixin.net.cn/documentation/#events) to notify us that it's done reloading. In particular, from reading bootstrap-table's docs/code, it seems like `onPostBody` might get called...

Reply:
I've fixed this issue by using Mutation Observers upon which arrive.js (https://github.com/uzairfarooq/arrive) is built, and executed the function for restoring pagination state only after pagination elements were added into the DOM.

Reply:
Sure. I've filed #27 to consider just reloading through bootstrap-table, as that seems simpler overall (and I think we'd get a nice `onLoadSuccess` event fire when the data has been loaded, no need for MutationObserver then), but for now we can go with this.